pkg/historical/service_test.go

```diff
@@ -83,39 +82,38 @@ var _ = Describe("BackFiller", func() {
 
 		It("Works for single block `ranges`", func() {
 			mockPublisher := &mocks.IterativeIPLDPublisher{
-				ReturnCIDPayload: []*eth.CIDPayload{mocks.MockCIDPayload},
-				ReturnErr:        nil,
+				ReturnErr: nil,
 			}
 			mockConverter := &mocks.IterativePayloadConverter{
-				ReturnIPLDPayload: []eth.ConvertedPayload{mocks.MockConvertedPayload},
+				ReturnIPLDPayload: []*eth.ConvertedPayload{&mocks.MockConvertedPayload},
 				ReturnErr:         nil,
 			}
-			mockRetriever := &mocks2.CIDRetriever{
+			mockRetriever := &mocks.Retriever{
 				FirstBlockNumberToReturn: 0,
-				GapsToRetrieve: []shared.Gap{
+				GapsToRetrieve: []eth.DBGap{
 					{
 						Start: 100, Stop: 100,
 					},
 				},
 			}
-			mockFetcher := &mocks2.PayloadFetcher{
-				PayloadsToReturn: map[uint64]shared.RawChainData{
+			mockFetcher := &mocks.PayloadFetcher{
+				PayloadsToReturn: map[uint64]statediff.Payload{
 					100: mocks.MockStateDiffPayload,
 				},
 			}
 			quitChan := make(chan bool, 1)
-			backfiller := &historical.BackFillService{
+			backfiller := &historical.Service{
 				Publisher:         mockPublisher,
 				Converter:         mockConverter,
 				Fetcher:           mockFetcher,
 				Retriever:         mockRetriever,
 				GapCheckFrequency: time.Second * 2,
 				BatchSize:         shared.DefaultMaxBatchSize,
-				BatchNumber:       shared.DefaultMaxBatchNumber,
+				Workers:           shared.DefaultMaxBatchNumber,
```

Comment:
what is value for ` shared.DefaultMaxBatchNumber`? can we run test case for:
1 block and 2 workers? doesn't it have any race condition?

Reply:
Right now `shared.DefaultMaxBatchNumber` is equal to 50, I should probably adjust that down and change the name  to `shared.DefaultNumberOfWorkers`. We have tested the concurrency in live staging but that would be a good unit test to include. I don't think I will add that here, this PR is primarily focused on the current refactoring to ipld-eth-indexer.